PurchasesTests/Networking/HTTPClientTests.swift

```diff
@@ -822,10 +822,8 @@ private extension HTTPClientTests {
                                                                                  options: []) as? [String: Any])
             return try XCTUnwrap(requestBodyDict["requestNumber"] as? Int)
         } catch {
-            XCTFail(error.localizedDescription)
```

Comment:
I meant to include the XCTFail though, so the result is the test failing instead of crashing.

Reply:
Ah ok, I hadn't understood what you meant 😓
![](https://media.giphy.com/media/vX9WcCiWwUF7G/giphy.gif)

BTW, WDYT about the message? Should we use the same message for both the XCTFail and the fatalError? 🤔

Reply:
If the `fatalError` executes the reason wouldn't be the `error` itself, but the fact that something happened with `XCTFail`. So I'd leave it as `"Unreachable"`.

Reply:
I found a problem with this code. What you mean it's this, right?
```swift
} catch {
    XCTFail("Couldn't extract the request number from the URLRequest")
    fatalError("Unreachable")
}
```

It seems that XCTFail doesn't interrupt the execution, so finally, the fatalError is reached. Am I missing something? 🤔

Reply:
Oh sorry! I thought it wouldn't. In that case I guess the right fix would be to make the return type `Int?` or `Int!` and return `nil`, to avoid the magic number.

Reply:
No problem. It seems to make the return type `Int?` and return `nil` works fine 👌🏼